src/rqt_image_view/ratio_layouted_frame.cpp

```diff
@@ -77,26 +77,48 @@ void RatioLayoutedFrame::resizeToFitAspectRatio()
   QRect rect = contentsRect();
 
   // reduce longer edge to aspect ration
-  double width = double(rect.width());
-  double height = double(rect.height());
-  if (width * aspect_ratio_.height() / height > aspect_ratio_.width())
+  double width;
+  double height;
+
+  if (outer_layout_)
+  {
+    width = outer_layout_->contentsRect().width() - 2;
+    height = outer_layout_->contentsRect().height() - 2;
```

Comment:
I think I was trying to center the image within the pre-scroll area widget.  As it is now I removed them and the image is not centered by much more than 2 pixels:

![image_in_scroll_area](https://cloud.githubusercontent.com/assets/1334122/25872262/06b69e22-34d8-11e7-918b-5ca161f324c0.png)

 (look at the bottom of the image)

What I would like to do is a separate pr that gets rid of that border and the oversized margins of some of the layouts- I think after that the image will be centered (or it won't be as obvious that it isn't), and much precious screen real estate will recovered as well.